src/main/scala/org/allenai/plugins/DeployPlugin.scala

```diff
@@ -1,90 +1,82 @@
-
-import com.typesafe.config.Config
-import com.typesafe.config.ConfigException
-import com.typesafe.config.ConfigFactory
-import com.typesafe.config.ConfigValueFactory
+import org.allenai.plugins.NodeJsPlugin.autoImport.{ NodeKeys, Npm }
+import com.typesafe.config.{ Config, ConfigException, ConfigFactory, ConfigValueFactory }
 import com.typesafe.sbt.SbtNativePackager._
 import com.typesafe.sbt.packager.archetypes.JavaAppPackaging
+import com.typesafe.sbt.packager.universal.UniversalPlugin
+import sbt._
+import sbt.Keys._
 
 import java.io.File
 
-import scala.collection.JavaConversions
-import scala.sys.process.Process
-import scala.sys.process.ProcessLogger
+import scala.collection.JavaConverters._
+import scala.sys.process.{ Process, ProcessLogger }
 import scala.util.Try
 
-/** Plugin to deploy a project to an ec2 instance. Handles copying binaries and
-  * config files, including setting up the correct per-environment override
-  * file, but does not handle restarting the remote application.
-  *
-  * This takes a path to a deploy configuration file as an argument, as well as
-  * a target within that configuration file to deploy. See global_deploy.conf
-  * for the list of required configuration values, and what they do.
+/** Plugin to deploy a project to an ec2 instance. Handles copying binaries and config files,
+  * generating a basic startup script, copying over an environment-specific config file if it
+  * exists, and restarting the service.
   *
-  * This will deploy from the current project in sbt.
+  * This looks for a file called `conf/deploy.conf` in your project's root directory for
+  * configuration. This should be a typesafe config file with any number of deploy environments
+  * configured.
   *
-  * This builds from the currently active git branch unless a version is
-  * specified, in which case it will check out that tag/branch/commit before
-  * building.
+  * This builds from the currently active git branch unless a version is specified, in which case it
+  * will check out that tag/branch/commit before building.
   *
-  * This will replace _completely_ the contents of the remote
-  * {bin,conf,lib,public} directories with the contents from the build. All other
-  * directories and files (such as pid and log files) will be left intact.
-  * After the new binaries are pushed, "${deploy.startup_script} restart" will
-  * be issued to restart the server.
+  * This will replace _completely_ the contents of the remote {bin,conf,lib,public} directories with
+  * the contents from the build. All other directories and files (such as pid and log files) will be
+  * left intact.  After the new binaries are pushed, "${deploy.startup_script} restart" will be
+  * issued to restart the server.
   *
-  * There is a special ~/.deployrc config file that is merged into any deploy
-  * target configs as an override.
+  * There is a special ~/.deployrc config file that is merged into any deploy target configs as an
+  * override.
```

Comment:
Yeah, although the reason for this was ssh keys, which have since moved into the AWS_PEM_FILE environment variable - and which probably won't be needed in the mid-term future anyway.

This is something I'd like to drop, but I'm trying not to make this PR a "cleanup everything" monolith.